app/src/main/java/org/cru/godtools/ui/tooldetails/ToolDetailsFragment.kt

```diff
@@ -131,15 +150,23 @@ class ToolDetailsFragment() :
         val code = tool?.code ?: return
         val locale = translation?.languageCode ?: return
 
-        activity?.run {
-            openToolActivity(code, tool.type, locale, showTips = true)
-            if (settings.getFeatureDiscoveredCount("$FEATURE_TUTORIAL_TIPS$code") < 3) {
-                settings.setFeatureDiscovered("$FEATURE_TUTORIAL_TIPS$code")
-                startTutorialActivity(
-                    PageSet.TIPS,
-                    Bundle().apply { putString(FORMAT_ARG_TOOL_NAME, translation.getName(tool).toString()) }
-                )
-            }
+        launchTrainingTips(code, tool.type, locale, false)
+    }
+
+    private fun launchTrainingTips(
+        code: String,
+        type: Tool.Type,
+        locale: Locale,
+        skipTutorial: Boolean
+    ) {
+        if (skipTutorial || settings.isFeatureDiscovered("$FEATURE_TUTORIAL_TIPS$code")) {
+            settings.setFeatureDiscovered("$FEATURE_TUTORIAL_TIPS${dataModel.tipsTool}")
```

Comment:
use `$code` instead of `${dataModel.tipsTool}`

Reply:
Okay, done!